<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>HFMainWindow</class>
 <widget class="QMainWindow" name="HFMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>929</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="baseSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="cg3::viewer::GLCanvas" name="glCanvas" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>929</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Rendering</string>
    </property>
    <addaction name="actionChange_Background_Color"/>
    <addaction name="actionFit_Scene"/>
    <addaction name="actionUpdate_Canvas"/>
    <addaction name="actionReset_Point_of_View"/>
    <addaction name="actionPerspective_Orthographic_Camera_Mode"/>
    <addaction name="actionShow_Axis"/>
    <addaction name="actionShow_Box"/>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="title">
     <string>Window</string>
    </property>
    <addaction name="actionFull_Screen"/>
    <addaction name="actionShow_Hide_Dock_Widget"/>
    <addaction name="actionShow_Hide_DrawList"/>
   </widget>
   <widget class="QMenu" name="menuFile_2">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionLoad_Mesh"/>
    <addaction name="actionLoad_HFD_Project"/>
    <addaction name="actionSave_HFD_Project"/>
    <addaction name="actionSave_HFD_Project_As"/>
    <addaction name="actionSave_Decomposition"/>
    <addaction name="actionSave_Packing"/>
    <addaction name="separator"/>
    <addaction name="actionLoad_Point_of_View"/>
    <addaction name="actionLoad_Point_Of_View_from"/>
    <addaction name="actionSave_Point_Of_View"/>
    <addaction name="actionSave_Point_Of_View_as"/>
    <addaction name="separator"/>
    <addaction name="actionSave_Snapshot"/>
   </widget>
   <addaction name="menuFile_2"/>
   <addaction name="menuFile"/>
   <addaction name="menuWindow"/>
  </widget>
  <widget class="QDockWidget" name="dockToolBox">
   <property name="sizePolicy">
    <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>300</width>
     <height>138</height>
    </size>
   </property>
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout"/>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockDrawList">
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_5">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QScrollArea" name="scrollArea">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>250</width>
         <height>0</height>
        </size>
       </property>
       <property name="layoutDirection">
        <enum>Qt::LeftToRight</enum>
       </property>
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>250</width>
          <height>445</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2"/>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="toolBar">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>647</width>
     <height>95</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>524287</width>
     <height>113</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::NoDockWidgetFeatures</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::AllDockWidgetAreas</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="spacing">
      <number>6</number>
     </property>
     <property name="leftMargin">
      <number>9</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QToolButton" name="loadToolButton">
       <property name="toolTip">
        <string>Load Mesh</string>
       </property>
       <property name="text">
        <string>Load Mesh</string>
       </property>
       <property name="icon">
        <iconset resource="../res/res.qrc">
         <normaloff>:/open-document-mesh.png</normaloff>:/open-document-mesh.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>48</width>
         <height>48</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="loadHFDToolButton">
       <property name="toolTip">
        <string>Load HFD Project</string>
       </property>
       <property name="text">
        <string>Load HFD Project</string>
       </property>
       <property name="icon">
        <iconset resource="../res/res.qrc">
         <normaloff>:/open-document-hfd.png</normaloff>:/open-document-hfd.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>48</width>
         <height>48</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="saveHFDToolButton">
       <property name="toolTip">
        <string>Save HFD Project</string>
       </property>
       <property name="text">
        <string>Save HFD Project</string>
       </property>
       <property name="icon">
        <iconset resource="../res/res.qrc">
         <normaloff>:/save-document-hfd.png</normaloff>:/save-document-hfd.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>48</width>
         <height>48</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="saveDecompositionToolButton">
       <property name="toolTip">
        <string>Save Decomposition Meshes</string>
       </property>
       <property name="text">
        <string>Save Decomposition</string>
       </property>
       <property name="icon">
        <iconset resource="../res/res.qrc">
         <normaloff>:/save-document-decomp.png</normaloff>:/save-document-decomp.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>48</width>
         <height>48</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="savePackingToolButton">
       <property name="toolTip">
        <string>Save Packing Meshes</string>
       </property>
       <property name="text">
        <string>save Packing</string>
       </property>
       <property name="icon">
        <iconset resource="../res/res.qrc">
         <normaloff>:/save-document-packing.png</normaloff>:/save-document-packing.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>48</width>
         <height>48</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="orthoToolButton">
       <property name="toolTip">
        <string>Toggle Orthographic Camera</string>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="../res/res.qrc">
         <normaloff>:/ortho.png</normaloff>:/ortho.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>48</width>
         <height>48</height>
        </size>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="showAxisToolButton">
       <property name="toolTip">
        <string>Show Axis</string>
       </property>
       <property name="text">
        <string>Show Axis</string>
       </property>
       <property name="icon">
        <iconset resource="../res/res.qrc">
         <normaloff>:/axis.png</normaloff>:/axis.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>48</width>
         <height>48</height>
        </size>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
       <property name="autoRaise">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="showBoxToolButton">
       <property name="toolTip">
        <string>Show Axis-Aligned Box</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../res/res.qrc">
         <normaloff>:/box.png</normaloff>:/box.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>48</width>
         <height>48</height>
        </size>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="rotationToolButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="toolTip">
        <string>Rotate Mesh</string>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="../res/res.qrc">
         <normaloff>:/rotation.png</normaloff>:/rotation.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>48</width>
         <height>48</height>
        </size>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="resetRotationToolButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="toolTip">
        <string>Reset Rotation</string>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="../res/res.qrc">
         <normaloff>:/reset_rotation.png</normaloff>:/reset_rotation.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>48</width>
         <height>48</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionSave_Snapshot">
   <property name="text">
    <string>Save Snapshot</string>
   </property>
  </action>
  <action name="actionShow_Axis">
   <property name="text">
    <string>Show Axis</string>
   </property>
   <property name="shortcut">
    <string>A</string>
   </property>
  </action>
  <action name="actionFull_Screen">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Full Screen</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
  </action>
  <action name="actionUpdate_Canvas">
   <property name="text">
    <string>Update Canvas</string>
   </property>
   <property name="shortcut">
    <string>U</string>
   </property>
  </action>
  <action name="actionFit_Scene">
   <property name="text">
    <string>Fit Scene</string>
   </property>
   <property name="shortcut">
    <string>F</string>
   </property>
  </action>
  <action name="actionChange_Background_Color">
   <property name="text">
    <string>Change Background Color</string>
   </property>
  </action>
  <action name="actionSave_Point_Of_View">
   <property name="text">
    <string>Save Point Of View</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionLoad_Point_of_View">
   <property name="text">
    <string>Load Point Of View</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionShow_Hide_Dock_Widget">
   <property name="text">
    <string>Show/Hide Dock Widget</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
  <action name="actionLoad_Point_Of_View_from">
   <property name="text">
    <string>Load Point Of View from...</string>
   </property>
  </action>
  <action name="actionSave_Point_Of_View_as">
   <property name="text">
    <string>Save Point Of View as...</string>
   </property>
  </action>
  <action name="actionShow_Hide_Console">
   <property name="text">
    <string>Show/Hide Console</string>
   </property>
   <property name="shortcut">
    <string>C</string>
   </property>
  </action>
  <action name="action2D_Mode">
   <property name="text">
    <string>2D Mode</string>
   </property>
   <property name="shortcut">
    <string>2</string>
   </property>
  </action>
  <action name="action3D_Mode">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>3D Mode</string>
   </property>
   <property name="shortcut">
    <string>3</string>
   </property>
  </action>
  <action name="actionReset_Point_of_View">
   <property name="text">
    <string>Reset Point of View</string>
   </property>
   <property name="shortcut">
    <string>R</string>
   </property>
  </action>
  <action name="actionPerspective_Orthographic_Camera_Mode">
   <property name="text">
    <string>Perspective/Orthographic Camera Mode</string>
   </property>
   <property name="shortcut">
    <string>O</string>
   </property>
  </action>
  <action name="actionToggle_Debug_Objects">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Toggle Debug Objects</string>
   </property>
   <property name="shortcut">
    <string>D</string>
   </property>
  </action>
  <action name="actionShow_Box">
   <property name="text">
    <string>Show Box</string>
   </property>
   <property name="shortcut">
    <string>B</string>
   </property>
  </action>
  <action name="actionShow_Hide_DrawList">
   <property name="text">
    <string>Show/Hide DrawList</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionLoad_Mesh">
   <property name="text">
    <string>Load Mesh</string>
   </property>
  </action>
  <action name="actionLoad_HFD_Project">
   <property name="text">
    <string>Load HFD Project</string>
   </property>
  </action>
  <action name="actionSave_HFD_Project">
   <property name="text">
    <string>Save HFD Project</string>
   </property>
  </action>
  <action name="actionSave_Decomposition">
   <property name="text">
    <string>Save Decomposition</string>
   </property>
  </action>
  <action name="actionSave_Packing">
   <property name="text">
    <string>Save Packing</string>
   </property>
  </action>
  <action name="actionSave_HFD_Project_As">
   <property name="text">
    <string>Save HFD Project As...</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>cg3::viewer::GLCanvas</class>
   <extends>QWidget</extends>
   <header location="global">cg3/viewer/glcanvas.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../res/res.qrc"/>
 </resources>
 <connections/>
</ui>
